upgrades/post/upgrade-4-6-0.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -euo pipefail
+
+if [ "$(yq '.spec.defaultRevision' "$ENV_DIR/env/apps/istio.yaml")" != "1-26-0" ]; then
+  echo "Checking Istio installation."
+  kubectl wait applications.argoproj.io -n argocd istio-system-istio-ingressgateway-public --timeout=600s --for=jsonpath='{.status.sync.status}'='Synced'
+  kubectl wait applications.argoproj.io -n argocd istio-system-istio-ingressgateway-public --timeout=600s --for=jsonpath='{.status.health.status}'='Healthy'
+  echo "Istio Ingress Gateway ready. Updating revision."
+  yq -I 4 -i '.spec.defaultRevision = "1-26-0"' "$ENV_DIR/env/apps/istio.yaml"
+  if [[ -z "$(kubectl get job --ignore-not-found -ojson -n default istio-operator-uninstall 2>/dev/null)" ]]; then
+    echo "Scheduling cleanup."
+    kubectl apply -f upgrades/post/upgrade-4-6-0-job.yaml
```

Comment:
What is the advantage of applying rest of the script as a Job?

Reply:
At this point, the values have barely been written to the values and not been committed yet. So for the time being, both the old and new gateway and routings are in place. Only the second pipeline run changes the default routings and shuts down the redundant gateway (which could take a while). The job waits for that condition without blocking the pipeline indefinitely. Also the job will retry the uninstall if the pod is killed, while the pipeline only leaves two options:
* Fail the pipeline if the uninstall fails. But then everything will need to run again.
* Step over the post-upgrade errors. But once the pipeline considers the upgrade done, it will never re-attempt to uninstall the operator if that failed.

Reply:
So it is more robust and non-blocking. Got it.